kernel/sync_block/src/condvar.rs

```diff
@@ -0,0 +1,107 @@
+use crate::MutexGuard;
+use cpu_local_preemption::hold_preemption;
+use mpmc_queue::Queue;
+use sync::DeadlockPrevention;
+use sync_spin::Spin;
+use task::{get_my_current_task, TaskRef};
+
+/// A condition variable.
+// TODO: Is there even a point to exposing this generic?
```

Comment:
Bump.

Reply:
see discord #sync-primitives discussion, but yea, we can remove it.